internal/controller/database_controller.go

```diff
@@ -75,15 +78,15 @@ var (
 	dbPhaseDelete               = "Deleting"
 )
 
-//+kubebuilder:rbac:groups=kinda.rocks,resources=databases,verbs=get;list;watch;create;update;patch;delete
-//+kubebuilder:rbac:groups=kinda.rocks,resources=databases/status,verbs=get;update;patch
-//+kubebuilder:rbac:groups=kinda.rocks,resources=databases/finalizers,verbs=update
-
+// +kubebuilder:rbac:groups=kinda.rocks,resources=databases,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=kinda.rocks,resources=databases/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=kinda.rocks,resources=databases/finalizers,verbs=update
 func (r *DatabaseReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	_ = r.Log.WithValues("database", req.NamespacedName)
-
 	reconcilePeriod := r.Interval * time.Second
 	reconcileResult := reconcile.Result{RequeueAfter: reconcilePeriod}
+
+	present := true
```

Comment:
Was think of another name. `present` is so generic. Also later for the function `HandleObjectModification` you use `isCreated` for it.

also for the function `createTemplatedSecrets`, it fells counter intuitive that this will then remove the item.

Reply:
1. I'm not even sure if I like passing `present` as an arg. Because function definitions are becoming huge. Maybe it should be another struct, or we can re-use something we already have. It's just an example
2. About the naming, I'm again not sure, what it should be. First, I've added `isCreated` but then I thought that databases are not created, when they are updated, so I've started calling it `present`. But I guess it's also not perfect, yes. I'll think about it
3. `createTemplatedSecret`. I don't think we should care about this function, because it's a deprecated feature anyway. We only need to make sure that it works until it's removed. But I also can rename it, so it's not that confusing. Also, it's not only creating, but removing as well, so maybe it makes sense to find a better name after all.